process/dataindexer/workItems/workItemBlock.go

```diff
@@ -42,14 +42,13 @@ func (wib *itemBlock) Save() error {
 	headerNonce := wib.outportBlockWithHeader.Header.GetNonce()
 	headerHash := wib.outportBlockWithHeader.BlockData.HeaderHash
 	shardID := wib.outportBlockWithHeader.Header.GetShardID()
-	defer func(startTime time.Time, headerHash []byte, headerNonce uint64, shardID uint32) {
-		log.Debug("wib.SaveBlockData",
-			"duration", time.Since(startTime),
-			"shardID", shardID,
-			"nonce", headerNonce,
-			"hash", headerHash,
-		)
-	}(time.Now(), headerHash, headerNonce, shardID)
+	startTime := time.Now()
+	defer log.Debug("wib.SaveBlockData",
```

Comment:
BUG. the duration is evaluated at this time, not at the end. 

proof:
```
func TestDefer(t *testing.T) {
	st := time.Now()
	defer log.Info("done", "elapsed time", time.Since(st))
	time.Sleep(2 * time.Second)
}
```
prints `INFO [2023-05-15 14:42:44.509]   done                                     elapsed time = 243ns `

while wrapping it into a function (like before) works:
```
func TestDeferLogWithinFunc(t *testing.T) {
	st := time.Now()
	defer func(startTime time.Time) {
		log.Info("done", "elapsed time", time.Since(st))
	}(st)
	time.Sleep(2 * time.Second)
}
```
prints `INFO [2023-05-15 14:43:59.349]   done                                     elapsed time = 2.000328239s `

so, please revert

Reply:
changed